core/src/draw_target/mod.rs

```diff
@@ -316,8 +319,11 @@ pub trait DrawTarget: Dimensions {
     /// The provided iterator is not required to provide `width * height` pixels to completely fill
     /// the area. In this case, `fill_contiguous` should return without error.
     ///
-    /// This method should not attempt to draw any pixels that fall outside the drawable area of the
-    /// target display. The `area` argument can be clipped to the drawable area using the
+    /// The provided iterator is not required to be limited to `width * height` pixels.
+    /// In this case, `fill_contiguous` should ignore the additional pixels and return without error.
+    ///
+    /// This method should ignore any pixels that fall outside the drawable area of the
+    /// target display without returning an error. The `area` argument can be clipped to the drawable area using the
```

Comment:
This is valid, but in this case extra care must be take. If `area` is clipped and the color iterator is not this can corrupt drawing.